core/src/main/java/org/apache/iceberg/hadoop/HadoopCatalog.java

```diff
@@ -276,8 +277,11 @@ public boolean dropNamespace(Namespace namespace) {
     }
 
     try {
-      return fs.delete(nsPath, false /* recursive */);
+      if (fs.listStatus(nsPath).length != 0) {
```

Comment:
`listStatus` returns an array which means it lists all the files before returning. Should we use `listFiles` which returns an Iterator. Iterator could be a lot more efficient and we can simply do `listFiles.hasNext`

Reply:
Good point! Sounds good to me. We can use `listStatusIterator`.